src/components/gr-button/gr-button.css

```diff
@@ -1,78 +1,246 @@
 :host {
-  display: inline-block;
+  display: var(--gr-button-display, inline-block);
   position: relative;
   width: auto;
   cursor: pointer;
 
-  --user-select: none;
+  --gr-button-padding-vertical: var(--gr-size-s);
+  --gr-button-padding-horizontal: var(--gr-size-l);
+  --gr-bg-color: var(--gr-color-primary);
+  --gr-border-color: var(--gr-color-primary);
+  --gr-content-color: var(--gr-color-content-primary);
+  --gr-button-font: var(--gr-font-title-3-medium);
 }
 
 .button {
   display: inline-flex;
-  align-items: stretch;
+  align-items: center;
   justify-content: center;
-  width: 100%;
-  height: var(--gr-button-height, 40px);
   border-style: solid;
   border-width: 1px;
-  border-radius: 4px;
+  border-radius: 6px;
   text-decoration: none;
   white-space: nowrap;
   vertical-align: middle;
-  padding: 12px 24px;
+  padding: var(--gr-button-padding-vertical) var(--gr-button-padding-horizontal);
   cursor: inherit;
-  background-color: var(--gr-color-primary);
-  border-color: var(--gr-color-primary);
-  color: #fff;
-  font: var(--gr-button-font, var(--gr-font-title-3-medium));
+  background-color: var(--gr-bg-color);
+  border-color: var(--gr-border-color);
+  color: var(--gr-content-color);
+  font: var(--gr-button-font);
 }
 
-:host([block]) {
-  display: block;
+:host([size='small']) {
+  --gr-button-font: var(--gr-font-title-4-medium);
+  --gr-button-padding-vertical: var(--gr-size-xs);
+  --gr-button-padding-horizontal: var(--gr-size-s);
+}
+
+:host([size='medium']) {
+  --gr-button-font: var(--gr-font-title-3-medium);
+  --gr-button-padding-vertical: var(--gr-size-s);
+  --gr-button-padding-horizontal: var(--gr-size-l);
+}
+
+:host([size='large']) {
+  --gr-button-font: var(--gr-font-title-3-medium);
+  --gr-button-padding-vertical: var(--gr-size-m);
+  --gr-button-padding-horizontal: var(--gr-size-xl);
 }
 
 .button:focus {
   outline: none;
 }
 
-.button:disabled {
+:host([link]) .button {
+  border: none;
+  background: none;
+  text-decoration: underline;
+
+  --gr-content-color: var(--gr-color-content-outline-primary);
+  --gr-button-font: var(--gr-font-title-3-medium);
+}
+
+:host([link]:hover:not([disabled])) .button {
```

Comment:
I think we can simplify this `:hover` styles by defining new variables for hover styles and setting them with variants. Like:

```css
.button {
    …
    —bl-button-hover-color: {primaryHoverColor};
}

.button:hover {
     background-color: var(—bl-button-hover-color);
}

:host([disabled]) .button {
     background-color: {disabledButtonBgColor};
}

:host([secondary]) {
   …
   —bl-button-hover-color: {secondaryHoverColor};
}
```

I expect this can reduce clutter a lot in CSS file.